src/common/browser-adapters/webextension-browser-adapter.ts

```diff
@@ -14,29 +17,37 @@ import browser, {
 import { BrowserAdapter } from './browser-adapter';
 import { CommandsAdapter } from './commands-adapter';
 import { StorageAdapter } from './storage-adapter';
-
 export abstract class WebExtensionBrowserAdapter
     implements BrowserAdapter, StorageAdapter, CommandsAdapter
 {
     constructor(private readonly browserEventManager: BrowserEventManager) {}
 
-    public initialize(browserEvents: DictionaryStringTo<Events.Event<any>>): void {
-        // Add browser listeners synchronously on initialization
-        for (const [eventType, event] of Object.entries(browserEvents)) {
-            this.browserEventManager.addBrowserListener(event, eventType);
-        }
-    }
-
     public abstract getManageExtensionUrl(): string;
 
+    public allRequiredEvents(): DictionaryStringTo<Events.Event<any>> {
```

Comment:
If we're moving the list of required events back here then we can remove the browser-event-provider.